driver/ppm_fillers.c

```diff
@@ -1269,16 +1270,43 @@ int f_proc_startupdate(struct event_filler_arguments *args)
 #else
 				exe_writable |= (inode_permission(file_inode(exe_file), MAY_WRITE) == 0);
 				exe_writable |= inode_owner_or_capable(file_inode(exe_file));
+#endif
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(3, 18, 0)
+				{
+					struct super_block *sb;
+					unsigned long sb_magic;
+
+					sb = exe_file->f_inode->i_sb;
+					sb_magic = sb->s_magic;
+					if(sb_magic == PPM_OVERLAYFS_SUPER_MAGIC)
+					{
+						struct dentry **upper_dentry;
+						struct inode **lower_inode;
+
+						upper_dentry = (struct dentry **)((char *)exe_file->f_inode + sizeof(struct inode));
+						lower_inode = (struct inode **)((char *)upper_dentry + sizeof(struct dentry *));
+
+						if(!*lower_inode && *upper_dentry)
+						{
+							exe_upper_layer = true;
+						}
+					}
+				}
 #endif
 			}
 #endif
+
 			fput(exe_file);
 		}
 
 		if (exe_writable) {
 			flags |= PPM_EXE_WRITABLE;
 		}
 
+		if (exe_upper_layer) {
```

Comment:
This is ok.
Like for the eBPF probe, i think you miss the `exe_upper_layer` in the `f_sched_prog_exec` filler (that is the new filler for exec for arm64 and s390x).